Sources/DiffableDataSource/DiffableTableViewDataSource+Replace.swift

```diff
@@ -10,9 +10,37 @@ import UIKit
 extension DiffableTableViewDataSource {
     // MARK: - Public API
 
-    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) throws {
-        try checkUniqueDiffableSections(sections: sections)
+    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) {
+        checkUniqueDiffableSections(sections: sections) { result in
+            switch result {
+            case let .success(sections):
+                self.unsafeReplace(sections: sections, animatingDifferences: animatingDifferences)
+
+            case let .failure(error):
+                Logger.shared.log(error: error)
+            }
+        }
+    }
+
+    public func checkUniqueDiffableSections(
+        sections: [DiffableSection<SectionID, RowID>],
+        result: (Result<[DiffableSection<SectionID, RowID>], DiffableError>) -> Void
+    ) {
+        let sectionIDs = sections.map { $0.identifier }
+        let rowIDs = sections.flatMap { $0.rows }
+
+        if sectionIDs.unique().count != sectionIDs.count {
+            result(.failure(DiffableError.sectionsNotUnique))
```

Comment:
I think we need to provide more context as part of this error.
If we received this as a fault report on Sentry would it have sufficient information to debug and fix?
(Can be done as a separate PR)